lib/bolt/plugin.rb

```diff
@@ -145,9 +143,7 @@ def self.setup(config, pal, analytics = Bolt::Analytics::NoopClient.new, **opts)
     DEFAULT_PLUGIN_HOOKS = { 'puppet_library' => { 'plugin' => 'puppet_agent', 'stop_service' => true } }.freeze
 
     attr_reader :pal, :plugin_context
-    attr_accessor :plugin_hooks
-
-    private_class_method :new
+    attr_writer :plugin_hooks
```

Comment:
Hmm, definitely seems safer to keep this but I also don't see where we write to it...

Reply:
I assumed it was for some tests way back or something. Also decided to not risk removing it, though. :P